mscore/preferences.cpp

```diff
@@ -38,6 +38,29 @@ void Preferences::init(bool storeInMemoryOnly)
             _settings = new QSettings();
             }
 
+#if defined(Q_OS_WIN)
+      // Use the correct default GUI font on Windows. Qt 5 and below use the deprecated function GetStockObject() with
+      // DEFAULT_GUI_FONT, which returns MS Shell Dlg 2 in 8 pt. MS Shell Dlg 2 is a virtual font that maps to Tahoma, which has
+      // not been the default Windows GUI font since 2006.
+      //
+      // The correct way to determine the default GUI font is to call SystemParametersInfoW() with SPI_GETNONCLIENTMETRICS and use
+      // the returned lfMessageFont structure. On all versions of Windows from Windows Vista through Windows 10, this typically
+      // returns Segoe UI in 9 pt.
+      //
+      // This problem is slated to be fixed in Qt 6. For details, see: https://bugreports.qt.io/browse/QTBUG-58610
+      //
+      // In the meantime, we can work around the problem by having Qt use the "QMessageBox" font instead, which is already being
+      // initialized the correct way.
+      QApplication::setFont(QApplication::font("QMessageBox"));
```

Comment:
Also would it be better to move this line into the `if()` statement below? Then it seems this fix wouldn't affect those installations which do not have this issue.

Reply:
No, these are two separate fixes.

Due to the bug in Qt, the default `QApplication` font will always be incorrect under Windows, no matter what is currently saved in the settings. If we moved this line into the `if` statement below, a fresh (or freshly factory-reset) installation of MuseScore would always be initialized with the wrong default font, which would render this fix useless.

The code inside the `if` statement is handling the specific case where the wrong default font has already been saved to the settings in the past. After we fix the default `QApplication` font, we check to see if the settings has the bad font saved; if so, we reset the font in the settings so that the correct default font (which we've just fixed with the `QApplication::setFont()` call) can be picked up.